### PR TITLE
Deprecate inventory plugin

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Deprecate inventory plugin

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/inventory/k8s.py [hunk 1]
```diff
@@ -19,6 +19,13 @@
       - Uses the kubectl connection plugin to access the Kubernetes cluster.
       - Uses k8s.(yml|yaml) YAML configuration file to set parameter values.
 
+    deprecated:
+      removed_in: 4.0.0
+      why: |
+        As discussed in U(https://github.com/ansible-collections/kubernetes.core/issues/31), we decided to
+        remove the k8s inventory plugin in release 4.0.0.
+      alternative: ""
```
Comment (gravesm) [inline plugins/inventory/k8s.py:27]:
```suggestion
      alternative: "Use M(kubernetes.core.k8s_info) and M(ansible.builtin.add_host) instead."
```

**Review by gravesm — APPROVED** [review]

**Review by abikouo — APPROVED** [review]

**Comment by gravesm** [comment]
regate

**Comment by gravesm** [comment]
regate

**Comment by gravesm** [comment]
regate

**Comment by gravesm** [comment]
regate

**Comment by gravesm** [comment]
regate

**Comment by abikouo** [comment]
regate

**Comment by tima** [comment]
This is the first step of #570.

**Pull request merged**